<a href="https://colab.research.google.com/github/amishakapse/Task1-Data-Decoder/blob/main/stacking_Model.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
import numpy as np
import pandas as pd

In [10]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [11]:
df = pd.read_csv("/content/drive/MyDrive/Letterbox Movie Classification Dataset.csv")

In [16]:
df.head(10)

Unnamed: 0                   Film_title           Director  Average_rating  \
0           0                      The Fan    Eckhart Schmidt            3.57   
1           1           Mad Max: Fury Road      George Miller            4.18   
2           2                     Suspiria      Dario Argento            3.93   
3           3          Lost in Translation      Sofia Coppola            3.79   
4           4                        Akira    Katsuhiro Otomo            4.28   
5           5               Things to Come    Mia Hansen-Løve            3.77   
6           6  Big Trouble in Little China     John Carpenter            3.72   
7           7                    Star Wars       George Lucas            4.17   
8           8                The Third Man         Carol Reed            4.25   
9           9                 The Big Sick  Michael Showalter            3.76   

                                         Genres  Runtime Original_language  \
0                           ['Horror', 'Drama']     92.0            German   
1    ['Adventure', 'Science Fiction', 'Action']    121.0           English   
2                                    ['Horror']     99.0           English   
3                ['Drama', 'Comedy', 'Romance']    102.0           English   
4    ['Animation', 'Action', 'Science Fiction']    124.0          Japanese   
5                                     ['Drama']    102.0            French   
6  ['Comedy', 'Adventure', 'Fantasy', 'Action']    100.0           English   
7    ['Adventure', 'Action', 'Science Fiction']    121.0           English   
8                       ['Mystery', 'Thriller']    105.0           English   
9                ['Drama', 'Comedy', 'Romance']    120.0           English   

                                         Description  \
0  The hero-worship that Simone has for a pop sin...   
1  An apocalyptic story set in the furthest reach...   
2  An American newcomer to a prestigious German b...   
3  Two lost souls visiting Tokyo -- the young, ne...   
4  A secret military project endangers Neo-Tokyo ...   
5  Nathalie teaches philosophy at a high school i...   
6  Truck driver Jack Burton gets embroiled in a s...   
7  Princess Leia is captured and held hostage by ...   
8  In postwar Vienna, Austria, Holly Martins, a w...   
9  Pakistan-born comedian Kumail Nanjiani and gra...   

                                             Studios  Watches  \
0                        ['Barbara Moorse Workshop']    12841   
1  ['Warner Bros. Pictures', 'Village Roadshow Pi...  2488721   
2                                ['Seda Spettacoli']   618171   
3           ['American Zoetrope', 'Elemental Films']  1596190   
4  ['MBS', 'Sumitomo Corporation', 'TOHO', 'Studi...   873068   
5  ['ARTE France Cinéma', 'Detailfilm', 'CG Ciném...    34777   
6  ['TAFT Entertainment Pictures', '20th Century ...   283873   
7             ['Lucasfilm Ltd.', '20th Century Fox']  3014320   
8                       ['London Films Productions']   228162   
9  ['FilmNation Entertainment', 'Apatow Productio...   382201   

   List_appearances    Likes   Fans  Lowest★  Medium★★★  Highest★★★★★  \
0             11032     3763     79      129       1660           714   
1            331681   927538  38000    12530     158356        511140   
2            220094   219548  13000     2814      53427         88628   
3            254180   493248  38000    15167     155110        193717   
4            175586   341567  21000     1822      40850        196532   
5             14370     8585    192       90       4274          1677   
6             65558    85788   3800     1250      30582         24938   
7            391415  1057816  30000     6041     161969        550396   
8            101285    70932   3400      351      11549         39520   
9             58859    93489    504     1410      36987         18997   

   Total_ratings  
0           9042  
1        1682389  
2         443757  
3        1076949  
4         600721  
5

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         10000 non-null  int64  
 1   Film_title         10000 non-null  object 
 2   Director           10000 non-null  object 
 3   Average_rating     10000 non-null  float64
 4   Genres             10000 non-null  object 
 5   Runtime            10000 non-null  float64
 6   Original_language  10000 non-null  object 
 7   Description        10000 non-null  object 
 8   Studios            10000 non-null  object 
 9   Watches            10000 non-null  int64  
 10  List_appearances   10000 non-null  int64  
 11  Likes              10000 non-null  int64  
 12  Fans               10000 non-null  int64  
 13  Lowest★            10000 non-null  int64  
 14  Medium★★★          10000 non-null  int64  
 15  Highest★★★★★       10000 non-null  int64  
 16  Total_ratings      1000

In [14]:
df.isnull().sum()

Unnamed: 0           0
Film_title           0
Director             0
Average_rating       0
Genres               0
Runtime              0
Original_language    0
Description          0
Studios              0
Watches              0
List_appearances     0
Likes                0
Fans                 0
Lowest★              0
Medium★★★            0
Highest★★★★★         0
Total_ratings        0
dtype: int64

In [15]:
df.describe()

Unnamed: 0  Average_rating       Runtime       Watches  \
count  10000.000000    10000.000000  10000.000000  1.000000e+04   
mean    4999.778800        3.268535    103.169352  1.703148e+05   
std     2887.257342        0.525744     38.070735  4.585329e+05   
min        0.000000        1.080000      2.000000  2.000000e+00   
25%     2499.750000        2.950000     89.000000  2.979750e+03   
50%     4999.500000        3.268535     99.000000  1.842500e+04   
75%     7500.250000        3.620000    113.000000  1.083580e+05   
max    10001.000000        4.710000    907.000000  5.195503e+06   

       List_appearances         Likes           Fans        Lowest★  \
count      10000.000000  1.000000e+04   10000.000000   10000.000000   
mean       26868.798600  4.373479e+04    1652.549700    1837.319600   
std        56751.328696  1.515314e+05    9368.258118    5617.492419   
min           33.000000  0.000000e+00       0.000000       0.000000   
25%         1674.000000  4.570000e+02       2.000000      35.000000   
50%         6503.000000  3.394000e+03      24.000000     181.000000   
75%        23908.250000  2.011225e+04     283.000000    1083.250000   
max       803670.000000  2.730813e+06  326000.000000  176637.000000   

           Medium★★★  Highest★★★★★  Total_ratings  
count   10000.000000  1.000000e+04   1.000000e+04  
mean    17887.078300  1.817158e+04   1.063923e+05  
std     43713.123082  8.323460e+04   2.947343e+05  
min         0.000000  0.000000e+00   0.000000e+00  
25%       422.750000  6.000000e+01   1.895000e+03  
50%      2422.500000  5.720000e+02   1.182300e+04  
75%     12727.000000  4.583250e+03   6.749875e+04  
max    629879.000000  2.010363e+06   4.091037e+06



*   This Dataset is Clean. There are no missing values.
*   The average runtime of the movie is 103 mins. The maximum runtime of a movie is 903 minutes, and the minimum is 2 minutes.
*   Average rating has mean 3.27 and a median is 3.27 which shows symmytrical distribution of movie ratings.



